WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -17,6 +17,7 @@ public class CoinsRegistry : ICoinsView
 		private HashSet<SmartCoin> SpentCoins { get; }
 		private HashSet<SmartCoin> LatestSpentCoinsSnapshot { get; set; }
 		private Dictionary<Script, Cluster> ClustersByScriptPubKey { get; }
+		private Dictionary<OutPoint, SmartCoin> CoinsByOutPoint { get; }
```

Comment:
This is incorrect, there can be many coins associated with one outpoint. It only works because in the tx processor we don't care about these coins.

Reply:
I fixed this.

Reply:
I saw this too late it seems. At this point it is not possible to have one output spent more than once, that's why I use a Dictionary tracking the spent output and the coin that spent it. 

In case the output is spent by a RBF transaction the spender coin (the one that spends the output) is updated and the data structure remains consistent at all time. 

I think if your assumption were correct then you should be able to write a unit test for that scenario, what i think is not possible.

Reply:
Testing the integrity of `CoinsByOutPoint` is possible. Sure the tx processing works anyway but the `CoinByOutPoint` would be a lie and would result in future confusion, as it's an incorrect association.